crates/rpc/rpc/src/call_guard.rs

```diff
@@ -0,0 +1,11 @@
+use std::sync::Arc;
+use tokio::sync::Semaphore;
+
+/// RPC Tracing call guard semaphore
+pub struct TracingCallGuard(Arc<Semaphore>);
```

Comment:
we need to derive clone here,
cloning this type also clones the Arc, which means multiple clones of `TracingCallGuard` are still using the same semaphore.